### ISSUE TITLE
Custom call back after create

### ISSUE THREAD

**Issue by Phocea** [marmelab/ng-admin#678]
Hello,

At the moment the FormController.js is expecting the correct response to be returned after a suscessful create.
i have returne payload of the form 
{"self": "http://localhost:8380/rest/api/v1/editorialChannels/ech10"}

the url constructed by the FormController.js is expecting an identifier value. Would be nice to be able to customize the POST action with the format of the response.
Can you document the format it is expecting at the moment, then I could transform the response in RestangularProvider.

Thanks

**Comment by abstratt** [comment]
@Phocea Fellow user here. The expected response is just an object with a slot per field, with the object identifier in a slot named 'id'. It will try to load that using the edit view, or the show view if the edit view is not enabled. How do you plan to populate the form without the data?

**Comment by Phocea** [comment]
@abstratt , I think you musunderstood me.
The commit fonction, is apparently using the POST response in order to pass the id identifier to the show/edit view to perform the corresponding GET that will populate the form.
My POST response is obvionsly not in the Ng-admin expected format. Once I work out what is expected I can modify the response using Restangular. Would be nicer to be able to provide a function for this on the create view config

**Comment by abstratt** [comment]
@Phocea I didn't realize it did a GET but I just tested it and you are right (because it just redirects to the show or edit view and those seem to always reload the data).

Is "ech10" your id there? If so, just adding a request interceptor that inserts an 'id' slot with the value extracted from the 'self' url would do (or it could rename 'self' to 'id').

I do something like that here:

https://github.com/abstratt/kirra/blob/master/kirra-ng-admin/admin.js#L234-L249

In my case, the id comes in a slot named 'objectId', so a need to copy that value to the resulting element.

**Comment by Phocea** [comment]
@abstratt Thanks for the pointer. I was doing something of the kind, but was modifying the response.data rather than the data itself. Your snippet put me back on track.

My point for approvement however would be the ability to set this transformation at the create view level using a function.
Depending on the resource create, the POST response may differ, so it would be more elegant to configure this on the view rather than handle each case in the RestangularProvider.addResponseInterceptor

**Comment by fzaninotto** [comment]
That's a recurrent discussion: Should ng-admin handle the API mapping, or should we leave it to Restangular and interceptors? We made the choice, a while ago, to focus on adding features on top of a standard API, and let the ability to map any API to this standard API using Restangular. This allows both a quick development, and a large user base. 

Besides, adding (back) the ability to customize API mapping at the ng-admin level means that there would be two ways to achieve the same thing... harder to learn and teach about. 

So until we've got a compelling reason to go back, we'll stick with the current responsibilities.
